config/src/main/java/org/opennms/integration/api/xml/ClasspathPollerConfigurationLoader.java

```diff
@@ -0,0 +1,304 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.integration.api.xml;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.opennms.integration.api.v1.config.poller.Downtime;
+import org.opennms.integration.api.v1.config.poller.Filter;
+import org.opennms.integration.api.v1.config.poller.Monitor;
+import org.opennms.integration.api.v1.config.poller.Package;
+import org.opennms.integration.api.v1.config.poller.Parameter;
+import org.opennms.integration.api.v1.config.poller.PollerConfigurationExtension;
+import org.opennms.integration.api.v1.config.poller.Range;
+import org.opennms.integration.api.v1.config.poller.Rrd;
+import org.opennms.integration.api.v1.config.poller.Service;
+import org.opennms.integration.api.xml.schema.poller.DowntimeXml;
+import org.opennms.integration.api.xml.schema.poller.FilterXml;
+import org.opennms.integration.api.xml.schema.poller.MonitorXml;
+import org.opennms.integration.api.xml.schema.poller.PackageXml;
+import org.opennms.integration.api.xml.schema.poller.ParameterXml;
+import org.opennms.integration.api.xml.schema.poller.PollerConfigurationXml;
+import org.opennms.integration.api.xml.schema.poller.RangeXml;
+import org.opennms.integration.api.xml.schema.poller.RrdXml;
+import org.opennms.integration.api.xml.schema.poller.ServiceXml;
+
+
+/**
+ * Used to load XML poller configuration from the class-path.
+ */
+public class ClasspathPollerConfigurationLoader extends ClasspathXmlLoader<PollerConfigurationXml> {
+
+    private final List<PollerConfigurationXml> configurations;
+    private  final List<Package> packages;
+    private final List<Monitor> monitors;
+
+
+    public ClasspathPollerConfigurationLoader(Class<?> clazz,  String subFolder, String ... fileNames) {
+        super(clazz, PollerConfigurationXml.class, subFolder, fileNames);
+        configurations = getObjects();
+        packages = configurations.stream()
+                .flatMap(c -> c.getPackages().stream())
+                .map(ClasspathPollerConfigurationLoader::mapXml)
+                .collect(Collectors.toUnmodifiableList());
+
+        monitors = configurations.stream()
+                .flatMap(c -> c.getMonitors().stream())
+                .map(ClasspathPollerConfigurationLoader::mapXml)
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
Please do not store this here but move this to the getters. If caching is required, you can cache the config object in the created `PollerConfigurationExtension` below.